src/generators/generator.js

```diff
@@ -50,14 +51,18 @@ class Generator {
   }
 
   componentPath() {
-    const fileName = normalizeCasing(this.componentName, this.fileCasing);
-    const compPath = `${this.componentDirPath()}/${fileName}.${this.fileExtension}`;
+    const fileBase = normalizeCasing(this.componentName);
+    let compPath = `${this.componentDirPath()}/`;
+    if (this.wrapFilesInFolders) compPath += `${fileBase}/`;
+    compPath += `${fileBase}.${this.fileExtension}`;
     return path.join(basePath, path.normalize(compPath));
   }
 
   componentTestPath() {
-    const fileName = normalizeCasing(this.componentName, this.fileCasing);
-    const testPath = `${this.testDirPath()}/${fileName}.test.${this.fileExtension}`;
+    const fileBase = normalizeCasing(this.componentName);
+    let testPath = `${this.testDirPath()}/`;
```

Comment:
missing second arg to `normalizeCasing`, pass in this.fileCasing as second arg on both the calls and it should work